### ISSUE TITLE
Feature: SEC-002 — Validate section heading order

### ISSUE THREAD

**Issue by nozomi-koborinai** [nozomi-koborinai/contextlint#30]
## Layer

Core (applicable to any structured Markdown project)

## Problem

SEC-001 validates that required sections **exist** in a document, but does not enforce their **order**. For template-driven documentation, the order of sections matters — e.g., "Overview" should always come before "Requirements", which should come before "Design".

Example:

```markdown
## Design        ← should come after Requirements
## Overview      ← should come first
## Requirements  ← should come after Overview
```

## Proposed Rule

| Property | Value |
|----------|-------|
| Rule ID | SEC-002 |
| Category | Section |
| Description | Sections must appear in the specified order |

### Config

```json
{
  "rule": "sec002",
  "options": {
    "order": ["Overview", "Requirements", "Design", "Test Plan"],
    "level": 2,
    "files": "docs/**/*.md"
  }
}
```

- `order`: Array of heading texts in the expected order
- `level`: (Optional) Only check headings at this level (1–6). When omitted, all heading levels are checked.
- `files`: (Optional) Glob pattern to limit which files are checked

### Behavior

- Only headings listed in `order` are checked — unlisted headings are ignored
- The rule checks **relative order**, not contiguity (other headings may appear between them)
- If a heading from `order` is missing entirely, SEC-002 does NOT report it (that is SEC-001's job)

### Example Output

```
docs/spec.md
  line 20   error  Section "Overview" must appear before "Design"  SEC-002
```

## Parser Changes

**Required.** The current `ParsedDocument.sections` is `string[]` (heading text only, no level info). The parser needs to expose heading levels:

```ts
// Add to ParsedDocument
headings: { text: string; level: number; line: number }[]
```

The existing `sections: string[]` field should be kept for backward compatibility with SEC-001.

## Why Core?

Section ordering is a universal concern for template-based documentation — RFCs, ADRs, design docs, runbooks. Not tied to any specific document framework.
